code/modules/projectiles/ammo_datums.dm

```diff
@@ -287,7 +287,6 @@
 
 		if(flags_ammo_behavior & AMMO_EXPLOSIVE)
 			execution_target.gib()
-		return TRUE
 	return ..()
```

Comment:
this whole proc can be early returned at the top with `if(!(flags_ammo_behavior & AMMO_HIGHIMPACT)) return ..()`

then keep the `return ..()` at the bottom and the whole thing is no longer indented and you remove the issue that created this code oversight bug